packages/app/pages/index.tsx

```diff
@@ -1,64 +1,265 @@
             <PageTitle title="" />
 
-            {(!modelCategories || modelCategories.length < 0) && <UnderMaintenance />}
-
-            {modelCategories?.map(category => (
-                <div key={category.name} className={styles.category}>
-                    <h3>{category.name}</h3>
-
-                    <div className={styles.models}>
-                        {category.models
-                            .sort((a, b) => a.name.localeCompare(b.name))
-                            .map(model => (
-                                <div key={model.name} className={styles.model}>
-                                    <Link href={`/${model.name}`}>
-                                        <Button
-                                            variant="light"
-                                            className="dashboard-model"
+            {!modelCategories.length && <UnderMaintenance />}
+
+            <article className={`${styles.card} rounded shadow-sm`}>
+                <h2 className="h3 m-0 ml-1">{`Recent Documents: ${filterDocumentBy}`}</h2>
+
+                <div className={`${styles.filter} mb-3`}>
+                    <label htmlFor="filter-document-state">Filter by:</label>
+                    <select
+                        id="filter-document-state"
+                        className="form-control"
+                        defaultValue={filterDocumentBy}
+                        onChange={e => setFilterDocumentBy(e.target.value)}
+                    >
+                        {userDocumentStates.map((state: string) => (
+                            <option value={state} key={state}>
+                                {state}
+                            </option>
+                        ))}
+                    </select>
+                </div>
+
+                {userRecentDocuments.map(document => {
+                    const { title, ['x-meditor']: meta } = document
+                    const { model, modifiedOn, state } = meta
+
+                    if (state !== filterDocumentBy) {
+                        return null
+                    }
+
+                    return (
+                        <div className="d-flex" key={title}>
+                            <Link
+                                href={`/${globalThis.encodeURIComponent(
+                                    model
+                                )}/${globalThis.encodeURIComponent(title)}`}
+                                className={styles.link}
+                            >
+                                <span className={styles.linkIconText}>
+                                    <ModelIcon
+                                        aria-hidden="true"
+                                        className="mr-3"
+                                        name={userIconMap[model].name}
+                                        color={userIconMap[model].color}
+                                    />
+                                    <span
```

Comment:
For reasons beyond me, browsers don't play well with CSS-truncated text and screen readers. We ARIA hide the truncated text and visually hide the full text. This works well for me on VoiceOver on Ventura.